src/psi_io.f90

```diff
@@ -56,7 +56,8 @@ module sds_def
         integer, dimension(mxdim) :: dims
         logical :: scale
         logical :: hdf32
-        type(rp1d), dimension(mxdim) :: scales
+      !   type(rp1d), dimension(mxdim) :: scales
+        real(REAL64), dimension(:,:), pointer :: scales
```

Comment:
Considering that we're just using 2D array like behavior of `scales`, I think it makes sense to have it as an allocatable instead of a pointer?